pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -551,13 +561,23 @@ public synchronized String toString(Calendar cal, Timestamp x,
     }
 
     cal = setupCalendar(cal);
-    cal.setTime(x);
+    long timeMillis = x.getTime();
+
+    // Round to microseconds
+    int nanos = x.getNanos();
+    if (nanos >= MAX_NANOS_BEFORE_WRAP_ON_ROUND) {
+      nanos = 0;
+      timeMillis++;
+    } else if (nanos % 1000 >= 500) {
+      nanos = nanos + 1000 - nanos % 1000;
```

Comment:
it appears appendTime now simply truncates at micro precision. that would seem to mean that `nanos += 1000;` would be sufficient here.